docs/analytics-ga4/ga4-event-tracker.md

```diff
@@ -72,16 +63,14 @@ In some situations we want to track dynamic elements like the 'Show/hide all sec
 
 It is also complicated by the fact that the JavaScript that creates this element is imported from `govuk-frontend` and therefore can't be modified directly. In order to get attributes onto this link, we have to do the following.
 
-- pass attributes to the accordion for the 'Show/hide all' link using the `data_attributes_show_all` option
-  - this should contain an object with one or many keys, each with a value converted into JSON
-  - this is appended to the main component element as `data-show-all-attributes`
+- Enable tracking for an accordion using the `ga4_tracking` option
+  - In the ruby template, `data-show-all-attributes` is appended to the main component element with the GA4 JSON
 - the 'Show/hide all' link is created by govuk-frontend JavaScript
 - the accordion JavaScript reads `data-show-all-attributes`
-  - it creates a `data-` attribute on the 'Show/hide all' link for each key in the JSON
-  - in the example above, the result will be `data-ga4="{ "event_name": "select_content", "type": "accordion" }"`
-- wrap the accordion in the event tracking script
-  - tracking will be activated by clicks on elements with a `data-ga4` attribute
-  - it checks for an `aria-expanded` attribute if the clicked element or parent element contains a `data-ga4-expandable` value or the `gem-c-accordion` class. If it detects one of those, it sets the `action` of the GA data accordingly.
+  - it moves the `data-show-all-attributes` JSON to the 'Show/hide all' link, renaming the attribute to `data-ga4-event`
+  - in the example above, the result will be `data-ga4-event="{ "event_name": "select_content", "type": "accordion", index: "0", index_total: "0" }"`
+  - tracking will be activated by clicks on elements with a `data-ga4-event` attribute
+  - it checks for an `aria-expanded` attribute, as the accordion will contains a `data-ga4-expandable` value. This sets the `action` of the GA data accordingly.
```

Comment:
Minor typo - "as the accordion will contain a `data-ga4-expandable` value"

Nitpick - would it be possible to say "This sets the `action` property of the `event_data` object accordingly."? I think this comes down to personal preference though so happy to leave it as it is if you prefer.

Reply:
Thanks @JamesCGDS, I've updated the docs